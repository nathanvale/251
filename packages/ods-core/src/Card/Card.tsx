import React, { ReactNode } from "react";
import {
  BreakpointVariants,
  SpaceVariants,
  Colors,
  CardPaddingVariants,
} from "@origin-digital/ods-types";
import { Box } from "../Box/Box";

export interface CardProps {
  padding?: CardPaddingVariants;
  "data-id"?: string;
  children?: ReactNode;
  backgroundColor?: keyof Pick<Colors, "transparent" | "grey50" | "white">;
}

type PaddingValues = Partial<Record<BreakpointVariants, SpaceVariants>>;

const cardPaddingForVariant: Record<
  CardPaddingVariants,
  {
    paddingX: PaddingValues;
    paddingY: PaddingValues;
  }
> = {
  small: {
    paddingX: {
      xs: "medium",
      sm: "xlarge",
    },
    paddingY: {
      xs: "large",
      lg: "xlarge",
    },
  },
  medium: {
    paddingX: {
      xs: "medium",
      sm: "xxlarge",
    },
    paddingY: {
      xs: "large",
      sm: "xxlarge",
    },
  },
  large: {
    paddingX: {
      xs: "medium",
      sm: "xxlarge",
      md: "xxxlarge",
    },
    paddingY: {
      xs: "large",
      sm: "xxlarge",
      md: "xxxlarge",
    },
  },
};

export const Card = ({
  "data-id": dataId,
  padding = "small",
  children,
  backgroundColor = "white",
}: CardProps) => (
  <Box
    width="full"
    backgroundColor={backgroundColor}
    data-id={dataId}
    {...cardPaddingForVariant[padding]}
  >
    {children}
  </Box>
);

Card.displayName = "Card";

Card.defaultProps = {
  "data-id": "card",
  padding: "small",
  backgroundColor: "white",
};
