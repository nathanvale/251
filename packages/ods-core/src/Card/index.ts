export { Card, defaultCardProps, cardPaddingForVariant } from "./Card";
export type CardProps = import("./Card").CardProps;
