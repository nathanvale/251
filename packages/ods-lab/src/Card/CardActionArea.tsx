import * as React from "react";
import MuiCardActionArea, {
  CardActionAreaProps as ICardActionArea,
} from "@material-ui/core/CardActionArea";

export interface CardActionAreaProps extends ICardActionArea {}

export const CardActionArea = (props: CardActionAreaProps) => (
  <MuiCardActionArea {...props}>{props.children}</MuiCardActionArea>
);
