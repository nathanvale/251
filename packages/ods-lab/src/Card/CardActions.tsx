import * as React from "react";
import MuiCardActions, {
  CardActionsProps as ICardActions,
} from "@material-ui/core/CardActions";

export interface CardActionsProps extends ICardActions {}

export const CardActions = (props: CardActionsProps) => (
  <MuiCardActions {...props}>{props.children}</MuiCardActions>
);
