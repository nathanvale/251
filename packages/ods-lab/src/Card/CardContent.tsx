import * as React from "react";
import MuiCardContent, {
  CardContentProps as ICardContent,
} from "@material-ui/core/CardContent";

export interface CardContentProps extends ICardContent {
  "data-id"?: string;
}

export const CardContent = (props: CardContentProps) => (
  <MuiCardContent {...props}>{props.children}</MuiCardContent>
);
