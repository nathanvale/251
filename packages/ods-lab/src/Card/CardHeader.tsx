import * as React from "react";
import MuiCardHeader, {
  CardHeaderProps as ICardHeader,
} from "@material-ui/core/CardHeader";

export interface CardHeaderProps extends ICardHeader {}

export const CardHeader = (props: ICardHeader) => (
  <MuiCardHeader {...props}>{props.children}</MuiCardHeader>
);
