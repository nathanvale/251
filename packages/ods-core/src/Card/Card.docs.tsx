/* eslint-disable react/display-name */
import React from "react";
import { ComponentDocs } from "../../../../docs/src/types";
import { Placeholder } from "../Placeholder/Placeholder";
import { CardProps, Card } from "..";

export const docs: ComponentDocs<CardProps> = {
  category: "Layout",
  componentName: "Card",
  description: "Cards provide content containers with responsive padding.",
  propDescriptions: {
    padding: "The size of the responsive paddingX and paddingY of the card.",
  },
  migrationGuide: false,
  examples: [
    {
      label: "Default Card",
      description: "Be default cards have a small padding size.",
      Code: () => (
        <Card>
          <Placeholder />
        </Card>
      ),
    },
    {
      label: "Medium Padding",
      Code: () => (
        <Card padding="medium">
          <Placeholder />
        </Card>
      ),
    },
    {
      label: "Large Padding",
      Code: () => (
        <Card padding="large">
          <Placeholder />
        </Card>
      ),
    },
  ],
};
