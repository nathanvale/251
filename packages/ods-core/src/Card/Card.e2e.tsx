import React from "react";
import { E2ETests } from "@origin-digital/ods-types";
import { Placeholder } from "../Placeholder/Placeholder";
import { Card } from "./Card";

export const tests: E2ETests = [
  {
    label: "Default card",
    Code: () => (
      <Card>
        <Placeholder />
      </Card>
    ),
    responsive: true,
  },
  {
    label: "Medium padding",
    Code: () => (
      <Card padding="medium">
        <Placeholder />
      </Card>
    ),
    responsive: true,
  },
  {
    label: "Large padding",
    Code: () => (
      <Card padding="large">
        <Placeholder />
      </Card>
    ),
    responsive: true,
  },
];
