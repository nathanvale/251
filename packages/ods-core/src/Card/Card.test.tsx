import React from "react";
import {render} from "@origin-digital/ods-testing-library";
import {Card} from "../Card/Card";

test("it renders a card", () => {
  const {container} = render(<Card />);
  expect(container.firstChild).toMatchSnapshot();
});

test("it renders a card with custom data-id", () => {
  const {container} = render(<Card data-id="myId" />);
  expect(container.firstChild).toHaveAttribute("data-id", "myId");
});

test("it renders a card with medium padding", () => {
  const {container} = render(<Card padding="medium" />);
  expect(container.firstChild).toMatchSnapshot();
});

test("it renders a card with large padding", () => {
  const {container} = render(<Card padding="large" />);
  expect(container.firstChild).toMatchSnapshot();
});

test("it renders a card with a selected background color", () => {
  const {container} = render(<Card padding="large" backgroundColor="grey4" />);
  expect(container.firstChild).toMatchSnapshot();
});
