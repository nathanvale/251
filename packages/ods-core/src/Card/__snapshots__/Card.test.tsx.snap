// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders a card 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  background-color: white;
  padding-top: 24px;
  padding-bottom: 24px;
  padding-left: 16px;
  padding-right: 16px;
  width: 100%;
}

@media screen and (min-width:992px) {
  .c0 {
    padding-top: 32px;
  }
}

@media screen and (min-width:992px) {
  .c0 {
    padding-bottom: 32px;
  }
}

@media screen and (min-width:576px) {
  .c0 {
    padding-left: 32px;
  }
}

@media screen and (min-width:576px) {
  .c0 {
    padding-right: 32px;
  }
}

<div
  class="c0"
  data-id="card"
  width="full"
/>
`;

exports[`it renders a card with a selected background color 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  background-color: #f8f8f8;
  padding-top: 24px;
  padding-bottom: 24px;
  padding-left: 16px;
  padding-right: 16px;
  width: 100%;
}

@media screen and (min-width:576px) {
  .c0 {
    padding-top: 48px;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    padding-top: 64px;
  }
}

@media screen and (min-width:576px) {
  .c0 {
    padding-bottom: 48px;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    padding-bottom: 64px;
  }
}

@media screen and (min-width:576px) {
  .c0 {
    padding-left: 48px;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    padding-left: 64px;
  }
}

@media screen and (min-width:576px) {
  .c0 {
    padding-right: 48px;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    padding-right: 64px;
  }
}

<div
  class="c0"
  data-id="card"
  width="full"
/>
`;

exports[`it renders a card with large padding 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  background-color: white;
  padding-top: 24px;
  padding-bottom: 24px;
  padding-left: 16px;
  padding-right: 16px;
  width: 100%;
}

@media screen and (min-width:576px) {
  .c0 {
    padding-top: 48px;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    padding-top: 64px;
  }
}

@media screen and (min-width:576px) {
  .c0 {
    padding-bottom: 48px;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    padding-bottom: 64px;
  }
}

@media screen and (min-width:576px) {
  .c0 {
    padding-left: 48px;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    padding-left: 64px;
  }
}

@media screen and (min-width:576px) {
  .c0 {
    padding-right: 48px;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    padding-right: 64px;
  }
}

<div
  class="c0"
  data-id="card"
  width="full"
/>
`;

exports[`it renders a card with medium padding 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  background-color: white;
  padding-top: 24px;
  padding-bottom: 24px;
  padding-left: 16px;
  padding-right: 16px;
  width: 100%;
}

@media screen and (min-width:576px) {
  .c0 {
    padding-top: 48px;
  }
}

@media screen and (min-width:576px) {
  .c0 {
    padding-bottom: 48px;
  }
}

@media screen and (min-width:576px) {
  .c0 {
    padding-left: 48px;
  }
}

@media screen and (min-width:576px) {
  .c0 {
    padding-right: 48px;
  }
}

<div
  class="c0"
  data-id="card"
  width="full"
/>
`;
